### PR TITLE
fix(dialog-controller): clear close promise when canDeactivate=>false

### PR DESCRIPTION
**Pull request by StrahilKazlachev**
fix for #189

### CODE REVIEW

File: src/dialog-controller.js [hunk 1]
```diff
@@ -58,7 +58,9 @@ export class DialogController {
    * @returns Promise An empty promise object.
    */
   close(ok: boolean, output?: any): Promise<DialogResult> {
-    if (this._closePromise) return this._closePromise;
+    if (this._closePromise) { 
+      return this._closePromise;
+    }
 
     this._closePromise = invokeLifecycle(this.viewModel, 'canDeactivate').then(canDeactivate => {
```
Comment (StrahilKazlachev) [inline src/dialog-controller.js]: I'm not sure what should be done if the promise returned from `invokeLifecycle` is rejected. Should `_closePromise` still be cleared. Or call `.error()`?

Comment (PWKad) [inline src/dialog-controller.js]: After reviewing this for a while I don't think it makes a difference so let's call `error` to be on the safe side.  Agreed?

Comment (StrahilKazlachev) [inline src/dialog-controller.js]: @PWKad When I wrote it I wasn't sure if unresolved promises cause memory leaks and that was my reasoning for wanting to call `.error()` - checked, don't cause.
Also `.error` will call `invokeLifecycle(this.viewModel, 'deactivate')`. Is it OK for `deactivate` to be called when `canDeactivate` has failed? If it's OK, I'll test it with calling `.error`.

**Comment by PWKad** [comment]
Gotcha I see what you mean now, if the promise throws I don't think calling `error` is proper like you mentioned.  Clearing it should be sufficient for now.  Long term might not hurt to have the state be a little bit easier to understand of why it's adding and clearing promises like that.

File: src/dialog-controller.js [hunk 2]
```diff
@@ -73,7 +75,10 @@ export class DialogController {
           });
       }
 
-      return Promise.resolve();
```
Comment (PWKad) [inline src/dialog-controller.js]: Was this not needed at all before?

Comment (StrahilKazlachev) [inline src/dialog-controller.js]: I think so. At least I do not see any meaning.
 1.  we are in the success callback
 2. `canDeactivate` is `false`
 3. we return
  - no `return` statement
  - `return Promise.resolve()` 
 4. `_closePromise` will be a promise that is resolved without a result

**Comment by EisenbergEffect** [comment]
@PWKad Is this ready to merge?

**Comment by StrahilKazlachev** [comment]
@PWKad ping

**Pull request merged**